pkg/vali/validation.go

```diff
@@ -0,0 +1,99 @@
+package vali
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+var mailRe = regexp.MustCompile(`\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z`)
+
+const tagName = "validate"
+
+type Validator interface {
+	Validate(interface{}) (bool, error)
+}
+type DefaultValidator struct {
+}
+
+func (v DefaultValidator) Validate(val interface{}) (bool, error) {
+	return true, nil
+}
+
+type StringValidator struct {
+	Min int
+	Max int
+}
+
+func (v StringValidator) Validate(val interface{}) (bool, error) {
+	l := len(val.(string))
```

Comment:
This will panic if val is not string. You should catch that with `v, ok := val.(string)`
But if this will always be a string, why not enforce it as a parameter?